### DISCUSSION TITLE
[Doc] document how to migrate from flannel to calico for existing clusters

### DISCUSSION THREAD

**Discussion by dberardo-com** [k3s-io/k3s#8304] [Ideas]
not just to start a new cluster from 0, but migrating an existing one.

**Comment by brandond** [comment]
We do not support changing the CNI on a running cluster, and do not plan on documenting it. I'll convert this to a discussion, if you do come up with some steps that work for you please post them.

**Reply by dberardo-com** [reply]
i see ... sorry to hear that ... i think the k3s doc should explicitly say "dont try to migrate an existing cluster to calico ... that is not supported!!"

i have managed to install calico on the cluster which previously had flunnel on it ... but this is quite a mess now, because DNS resolution is not working quite right ...

thing is ... i am not able to revert back to flannel now, because the nodes are tainted with "NetworkUnavailable" if all workloads of calico are down ... 

so i am quite stuck.

perhaps i should give this migration procedure a try ?? 

https://docs.tigera.io/calico/latest/getting-started/kubernetes/flannel/migration-from-flannel

**Reply by mmuto118** [reply]
@dberardo-com 
Hi,
Thank you for posting this topic.
I am planning to maigrate from flannel to calico on our runing cluster.
The main reason for this is to get real source ip address by using traefik(Ingress) in calico networking.
you try the bellow procudure?
https://docs.tigera.io/calico/latest/getting-started/kubernetes/flannel/migration-from-flannel

**Reply by brandond** [reply]
The normal way to get real source address in pods behind a LoadBalancer service is to set ExternalTrafficPolicy=Local:
https://kubernetes.io/docs/tasks/access-application-cluster/create-external-load-balancer/#preserving-the-client-source-ip

Note that changing the CNI isn't required, these steps will actually work with traefik, flannel, and servicelb.

**Reply by mmuto118** [reply]
@brandond 
Thank you for reply.
your advice helps a lot.
I will try the link proveded by you.

**Reply by jerome7885** [reply]
I beginner in kubernetes, but :
With K3s flannel and ingress-nginx, set ExternalPolicyNetwork at Local. No, problem with the Layer, the problem is Klipper, i know.
Sorry for my bad english. Mais j'ai vraiment tout essayé depuis 5 jours.

**Reply by matschen** [reply]
@mmuto118 Hello, I’d like to ask whether your issue has been resolved. I’m facing the same situation as well. Thank you!
